basis/cpu/x86/64/unix/unix.factor

```diff
@@ -24,16 +24,27 @@ M: x86.64 reserved-stack-space 0 ;
         [ 8 mod zero? [ t , ] when , ] assoc-each
     ] { } make { t } split harvest ;
 
-: flatten-small-struct ( c-type -- seq )
-    struct-types&offset split-struct [
+:: flatten-small-struct ( c-type -- seq )
+    c-type struct-types&offset split-struct [
         [ lookup-c-type c-type-rep reg-class-of ] map
         int-regs swap member? int-rep double-rep ?
         f f 3array
-    ] map ;
+    ] map :> reps
+    int-reg-reps get float-reg-reps get and [
+        reps [ first int-rep? ] count :> int-mems
```

Comment:
If you made ``reg-reps`` like above, this could just be:

```factor
reps reg-reps :> ( int-mems float-mems )
```

Looking at it in more detail, in ``reg-reps`` you check for ``second not`` but here you don't... 

Bug?  Or not possible code path?

Reply:
The second element of the `reps` element is all `f`, so it was not checked here.
Even if the check is done, I don't think it will affect the operation. (Strictly speaking, it affects the execution speed.)